src/plugin/emulate_speaker_via_audio_plugin.c

```diff
@@ -53,6 +53,11 @@ void push_audio_samples_via_audio_plugin(void* user_data, const void* buffer, si
 
     /* exploit the fact that buffer points in g_dev.ri.rdram.dram to retreive dram_addr_reg value */
     ai->regs[AI_DRAM_ADDR_REG] = (uint8_t*)buffer - (uint8_t*)ai->ri->rdram.dram;
+    if (ai->fixed_audio_pos) {
```

Comment:
This logic shouldn't be here as it is only meant as a "compatibility layer" to use zilmar spec audio plugins. Not a core behavior. This should go in write_ai_regs. Just add a case AI_DRAM_ADDR_REG in there and these lines.